packages/salesforcedx-apex-debugger/src/index.ts

```diff
@@ -26,3 +26,19 @@ export interface WorkspaceSettings {
   connectionTimeoutMs: number;
   setBreakpointTimeoutMs: number;
 }
+
+export interface Metric {
+  subject: string;
+  type: string;
+}
+
+// Type guard to check if the object conforms to Metric
+export const isMetric = (input: unknown): input is Metric => (
+  input !== null &&
+  typeof input === 'object' &&
+  Object.keys(input).length === 2 &&
+  'subject' in input &&
+  typeof input.subject === 'string' &&
+  'type' in input &&
+  typeof input.type === 'string'
+);
```

Comment:
```suggestion
export const isMetric = (input: unknown): input is Metric => (
  !!input &&
  Object.keys(input).every(key => ['subject', 'type'].includes(key)) &&
  Object.values(input).every(value => typeof value === 'string')
);
```

Reply:
Wow, this is very consolidated compared to what was there before!  I really would not have thought this is how to do the improvement!  Thank you @peternhale!